base/osutils.jl

```diff
@@ -2,59 +2,95 @@
 
 const OS_NAME = ccall(:jl_get_OS_NAME, Any, ())
 
-function is_unix(os::Symbol)
-    if (os==:Windows) return false;
-    elseif (os==:Linux) return true;
-    elseif (os==:FreeBSD) return true;
-    elseif (os==:Darwin) return true;
-    else throw(ArgumentError("unknown operating system, $(repr(os))"))
-    end
-end
+"""
+    is_unix([os])
 
-function _os_test(qm,ex,test)
-    @assert qm == :?
-    @assert isa(ex,Expr)
-    @assert ex.head == :(:)
-    @assert length(ex.args) == 2
-    if test
-        return esc(ex.args[1])
+Predicate for testing if the OS provides a Unix-like interface.
+See documentation :ref:`Handling Operating System Variation <man-handling-operating-system-variation>`\ .
+"""
+function is_unix(os::Symbol = OS_NAME)
+    if is_windows(os)
+        return false
+    elseif is_linux(os) || is_bsd(os)
+        return true
     else
-        return esc(ex.args[2])
+        throw(ArgumentError("unknown operating system \"$os\""))
```

Comment:
What's the exception garentee about `@pure` functions? If `is_*` can be `@pure` I believe a lot of the `@static` won't be necessary. (this also determines if we can make math functions like `sin(::Float64)` pure)